### ISSUE TITLE
Fix to speed up math operations by more than 50%

### ISSUE THREAD

**Issue by kempeng** [shopspring/decimal#136]
see issue #128:
fix for decimal.go/rescale()

```
// NOTE(vadim): must convert exps to int64 before - to prevent overflow
	diff := abs64(int64(exp) - int64(d.exp))
// NEW TEST, avoids unnecessary rescaling
	if diff == 0 {
		return d
	}
// END
	value := new(big.Int).Set(d.value)
	expScale := new(big.Int).Exp(tenInt, big.NewInt(int64(diff)), nil)
```

**Comment by mwoss** [comment]
Thanks for pointing this out @kempeng. I will closer on this and continue discussion in mentioned issue #128

**Comment by mwoss** [comment]
Closed by #160
